pkg/aws/awscluster/mutate_awscluster.go

```diff
@@ -0,0 +1,82 @@
+// Package awsmachinedeployment intercepts write activity to AWSMachineDeployment objects.
+package awscluster
+
+import (
+	"fmt"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	admissionv1 "k8s.io/api/admission/v1"
+
+	"github.com/giantswarm/aws-admission-controller/config"
+	"github.com/giantswarm/aws-admission-controller/pkg/mutator"
+)
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+// Mutator for AWSMachineDeployment object.
+type Mutator struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+
+	podCIDRBlock string
+}
+
+func NewMutator(config config.Config) (*Mutator, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	var cidrBlock string = fmt.Sprintf("%s/%s", config.PodSubnet, config.PodCIDR)
+	mutator := &Mutator{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+
+		podCIDRBlock: cidrBlock,
+	}
+
+	return mutator, nil
+}
+
+// Mutate is the function executed for every matching webhook request.
+func (m *Mutator) Mutate(request *admissionv1.AdmissionRequest) ([]mutator.PatchOperation, error) {
+	var result []mutator.PatchOperation
+
+	if request.DryRun != nil && *request.DryRun {
```

Comment:
Dry run looks more like a pass through. What is the idea behind this feature?

Reply:
If I remember correctly the dry run was added when a customer experienced problems due to the dry run flag not being available. Maybe @paurosello remembers better than I do?